wp-content/plugins/core/src/Blocks/Types/Hero/Model.php

```diff
@@ -32,23 +25,36 @@ private function get_layout() {
 	/**
 	 * @return int|bool
 	 */
-	private function get_media() {
+	private function get_media_id() {
 		return $this->get( Hero::IMAGE, false );
 	}
 
 	/**
 	 * @return array
 	 */
-	private function get_content(): array {
+	private function get_cta_args(): array {
+		$cta = wp_parse_args( $this->get( Hero::CTA, [] ), [
+			'title'  => '',
+			'url'    => '',
+			'target' => '',
+		] );
+
+		return [
+			'content' => $cta[ 'title' ],
+			'url'     => $cta[ 'url' ],
+			'target'  => $cta[ 'target' ],
+		];
+	}
+
+	/**
+	 * @return array
+	 */
+	private function get_content_args(): array {
 		return [
-			self::LEAD_IN => $this->get( Hero::LEAD_IN, '' ),
-			self::TITLE   => $this->get( Hero::TITLE, '' ),
-			self::TEXT    => $this->get( Hero::DESCRIPTION, '' ),
-			self::CTA     => wp_parse_args( $this->get( Hero::CTA, [] ), [
-				'title'  => '',
-				'url'    => '',
-				'target' => '',
-			] ),
+			Hero::LEAD_IN     => $this->get( Hero::LEAD_IN, '' ),
+			Hero::TITLE       => $this->get( Hero::TITLE, '' ),
+			Hero::DESCRIPTION => $this->get( Hero::DESCRIPTION, '' ),
+			Hero::CTA         => $this->get_cta_args(),
```

Comment:
Which also means we either need to ensure when we spin up new blocks that these constants map to the component args (and likely hoist are commonly used field groups like for the block header) OR default to using the straight component strings as the keys and only leverage the constants for the `get`. Thoughts?

cc @dpellenwood @jbrinley

Reply:
My vote would be the latter so there is no confusion (along with creating commonly used field groups)

Reply:
I'm with you on this. I cleaned it up a bit. I'm making the component take specific arguments for each part of the content-block instead of an array of data.